File: Modules/HumanResource/EmployeeProfile/MasterDeduction/MasterDeductionForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.MasterDeduction")]
[BasedOnRow(typeof(MasterDeductionRow), CheckNames = true)]
public class MasterDeductionForm
{
    [HalfWidth]
    public string DeductionCode { get; set; }

    [HalfWidth]
    public double Amount { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
    [Category("Deduction Frequency")]
    [QuarterWidth]
    public int Recurring { get; set; }
    [QuarterWidth]
    public int OneTime { get; set; }
}

File: Modules/ServerTypes/EmployeeProfile/MasterDeductionForm.ts
import { StringEditor, DecimalEditor, TextAreaEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface MasterDeductionForm {
    DeductionCode: StringEditor;
    Amount: DecimalEditor;
    Description: TextAreaEditor;
    Recurring: BooleanEditor;
    OneTime: BooleanEditor;
}

export class MasterDeductionForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.MasterDeduction';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MasterDeductionForm.init)  {
            MasterDeductionForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;

            initFormType(MasterDeductionForm, [
                'DeductionCode', w0,
                'Amount', w1,
                'Description', w2,
                'Recurring', w3,
                'OneTime', w3
            ]);
        }
    }
}